pkg/result/ignore_test.go

```diff
@@ -0,0 +1,76 @@
+package result
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseIgnoreFile(t *testing.T) {
+	t.Run("happy path valid config file", func(t *testing.T) {
+		got, err := ParseIgnoreFile(context.TODO(), "testdata/.trivyignore")
+		require.NoError(t, err)
+		assert.Equal(t, "testdata/.trivyignore", got.FilePath)
+
+		// IDs in .trivyignore are treated as IDs for all scanners
+		// as it is unclear which type of security issue they are
+		assert.Len(t, got.Vulnerabilities, 6)
+		assert.Len(t, got.Misconfigurations, 6)
+		assert.Len(t, got.Secrets, 6)
+		assert.Len(t, got.Licenses, 6)
+	})
+
+	t.Run("happy path valid YAML config file", func(t *testing.T) {
+		got, err := ParseIgnoreFile(context.TODO(), "testdata/.trivyignore.yaml")
+		require.NoError(t, err)
+		assert.Equal(t, "testdata/.trivyignore.yaml", got.FilePath)
+		assert.Len(t, got.Vulnerabilities, 5)
+		assert.Len(t, got.Misconfigurations, 3)
+		assert.Len(t, got.Secrets, 3)
+		assert.Len(t, got.Licenses, 1)
+	})
+
+	t.Run("empty YAML file passed", func(t *testing.T) {
+		f, err := os.CreateTemp("", "TestParseIgnoreFile-*.yaml")
+		require.NoError(t, err)
+		defer os.Remove(f.Name())
+
+		_, err = ParseIgnoreFile(context.TODO(), f.Name())
+		require.NoError(t, err)
+	})
+
+	t.Run("invalid YAML file passed", func(t *testing.T) {
+		f, err := os.CreateTemp("", "TestParseIgnoreFile-*.yaml")
+		require.NoError(t, err)
+		defer os.Remove(f.Name())
+		_, _ = f.WriteString("this file is not a yaml file")
+
+		got, err := ParseIgnoreFile(context.TODO(), f.Name())
+		assert.Contains(t, err.Error(), "yaml decode error")
+		assert.Empty(t, got)
+	})
+
+	// TODO(simar7): This test currently fails as we don't validate
+	// the correctness of ignore file when not in YAML format
```

Comment:
Yes I added a test for it https://github.com/aquasecurity/trivy/blob/e12cb8849915d320381f7957eb85db4b852b14c6/pkg/result/ignore_test.go#L45-L54

We don't validate the correctness of non yaml ignore files. We silently return with no error.

Reply:
Ah, I understand. Since this is intended, we should change the test "invalid file passed" to expect no error. If we don't think this is the correct behavior, we should change the logic.

Reply:
OK I updated the test but I am not sure if this current behavior is correct per se. The user can pass an incorrect ignore config (non yaml) and realize that their ignores aren't effective, leading to confusion.

Let's tackle it in a separate PR as a feature improvement.

Reply:
It may be better to return an error, like "the ignore yaml must end with `*.yaml`, if the file doesn't have the extension. But yes, we can do that in another PR.

Reply:
I think we should eliminate the support for non yaml formatted ignore files altogether. There's a lot of cusstom parsing logic https://github.com/aquasecurity/trivy/blob/1f5f34895823fae81bf521fc939bee743a50e304/pkg/result/ignore.go#L236-L269  that it needs and I don't see any value supporting it https://trivy.dev/v0.57/docs/configuration/filtering/#trivyignore 

It is also marked experimental so it isn't a breaking change to remove that format.

Reply:
`.trivyignore.yaml` is experimental, but `.trivyignore` is not. I'm sure many users depend on it. It's very challenging to drop the support.